app/src/main/java/ch/hikemate/app/ui/map/MapScreen.kt

```diff
@@ -0,0 +1,271 @@
+            }
+          }
+        }
+      },
+      sheetPeekHeight = 400.dp) {}
+}
+
+@Composable
+fun HikingRouteItem(
+    title: String,
+    altitudeDifference: Int,
+    isSuitable: Boolean,
+    onClick: () -> Unit,
+    modifier: Modifier = Modifier
+) {
+  Row(
+      modifier =
+          modifier
+              .fillMaxWidth()
+              .clickable(onClick = onClick)
+              .padding(16.dp, 8.dp)
+              .background(MaterialTheme.colorScheme.surface, RoundedCornerShape(8.dp))
+              .testTag(MapScreen.TEST_TAG_HIKE_ITEM),
+      verticalAlignment = Alignment.CenterVertically) {
+        Column(modifier = Modifier.weight(1f)) {
+          Text(
+              text = title,
+              style = MaterialTheme.typography.titleLarge,
+              fontWeight = FontWeight.Bold)
+          Spacer(modifier = Modifier.height(8.dp))
+
+          Row(
+              verticalAlignment = Alignment.CenterVertically,
+              modifier = Modifier.fillMaxWidth().height(IntrinsicSize.Min)) {
+                Box(
+                    modifier =
+                        Modifier.weight(1f)
+                            .fillMaxHeight()
+                            .background(
+                                MaterialTheme.colorScheme.primary.copy(alpha = 0.1f),
+                                RoundedCornerShape(4.dp)))
+
+                Spacer(modifier = Modifier.width(8.dp))
+
+                Column {
+                  Text(text = "Altitude difference", style = MaterialTheme.typography.bodySmall)
+                  Text(
+                      text = "${altitudeDifference}m",
+                      style = MaterialTheme.typography.bodyLarge,
+                      fontWeight = FontWeight.Bold)
+                }
+              }
+
+          Spacer(modifier = Modifier.height(4.dp))
+
+          Row(verticalAlignment = Alignment.CenterVertically) {
+            Icon(
+                imageVector = if (isSuitable) Icons.Default.Check else Icons.Default.Warning,
+                // The icon is only decorative, the following message is enough for accessibility
+                contentDescription = null,
+                // TODO : Replace suitable and challenging icon colors with theme colors
+                tint = if (isSuitable) Color(0xFF4CAF50) else Color(0xFFFFC107),
+                modifier = Modifier.size(16.dp))
+            Spacer(modifier = Modifier.width(4.dp))
+            Text(
+                text =
+                    if (isSuitable)
+                        LocalContext.current.getString(R.string.map_screen_suitable_hike_label)
+                    else LocalContext.current.getString(R.string.map_screen_challenging_hike_label),
+                style = MaterialTheme.typography.bodySmall,
+                // TODO : Replace suitable and challenging icon colors with theme colors
+                color = if (isSuitable) Color(0xFF4CAF50) else Color(0xFFFFC107))
+          }
+        }
```

Comment:
Praise: I like your treatment of insuitable hikes, changing the colors, adding a warning symbol and changing the text!

Reply:
Thanks !